tests/tensorlake/test_graph_metadata.py

```diff
@@ -0,0 +1,248 @@
+            description="test",
+            start_node=function_with_custom_retries,
+            retries=Retries(
+                max_retries=3,
+                initial_delay=10.0,
+                max_delay=120.0,
+                delay_multiplier=10.0,
+            ),
+        )
+        graph_metadata: ComputeGraphMetadata = graph.definition()
+        retry_policy_metadata: RetryPolicyMetadata = (
+            graph_metadata.start_node.compute_fn.retry_policy
+        )
+        self.assertIsNotNone(retry_policy_metadata)
+        self.assertEqual(retry_policy_metadata.max_retries, 2)
+        self.assertEqual(retry_policy_metadata.initial_delay_sec, 2.0)
+        self.assertEqual(retry_policy_metadata.max_delay_sec, 10.0)
+        self.assertEqual(retry_policy_metadata.delay_multiplier, 5.0)
+
+
+class TestGraphMetadataFunctionResources(unittest.TestCase):
+    def test_default_function_resources(self):
+        @tensorlake_function()
+        def function_without_resources(x: int) -> str:
+            return "success"
+
+        graph = Graph(
+            name=test_graph_name(self),
+            description="test",
+            start_node=function_without_resources,
+        )
+        graph_metadata: ComputeGraphMetadata = graph.definition()
+        resource_metadata: ResourceMetadata = (
+            graph_metadata.start_node.compute_fn.resources
+        )
+        self.assertIsNotNone(resource_metadata)
+        self.assertEqual(resource_metadata.cpus, 0.125)
+        self.assertEqual(resource_metadata.memory_mb, 128)
+        self.assertEqual(resource_metadata.ephemeral_disk_mb, 100 * 1024)
+        self.assertIsNone(resource_metadata.gpu)
+
+    def test_custom_function_resources(self):
+        @tensorlake_function(
+            resources=Resources(
+                cpu=2.25,
+                memory=2048,
+                ephemeral_disk=10,
+                gpu=GPU_MODEL.H100,
+            )
+        )
+        def function_with_resources(x: int) -> str:
+            return "success"
+
+        graph = Graph(
+            name=test_graph_name(self),
+            description="test",
+            start_node=function_with_resources,
+        )
+        graph_metadata: ComputeGraphMetadata = graph.definition()
+        resource_metadata: ResourceMetadata = (
+            graph_metadata.start_node.compute_fn.resources
+        )
+        self.assertIsNotNone(resource_metadata)
+        self.assertEqual(resource_metadata.cpus, 2.25)
+        self.assertEqual(resource_metadata.memory_mb, 2048)
+        self.assertEqual(resource_metadata.ephemeral_disk_mb, 10 * 1024)
+        self.assertIsNotNone(resource_metadata.gpu)
+        gpu_metadata: GPUResourceMetadata = resource_metadata.gpu
+        self.assertEqual(gpu_metadata.count, 1)
+        self.assertEqual(gpu_metadata.model, GPU_MODEL.H100)
+
+    def test_custom_function_resources_many_gpus(self):
+        @tensorlake_function(
+            resources=Resources(
```

Comment:
Lets make this flat - 
```
@tensorlake_function(gpu="A100-40GB:4", cpu=0.8, memory=2)
```
And make GB the unit for memory, and mb and Kbs can be expressed as fractions. 

We can read these attributes and create the `Resources` object you have defined and put it in the GraphMetadata object that we send to the server.

Reply:
Sounds good. Implemented in the new PR version, ptal.